markdownlint/TOP010_useLazyNumbering/TOP010_useLazyNumbering.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
+  names: ["TOP010", "ordered-list-for-lazy-numbering"],
+  description: "Ordered lists should use lazy numbering",
+  information: new URL(
+    "https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP010.md"
+  ),
+  tags: ["lazy-numbering"],
+  parser: "markdownit",
+  function: function TOP010(params, onError) {
+    params.parsers.markdownit.tokens.forEach((token) => {
+      if (
+        token.tag === "li" &&
+        token.line !== undefined &&
+        /\d/.test(token.line) &&
+        !/^\s*1\./.test(token.line)
+      ) {
+        const originalText = token.line;
+        // https://regexr.com/80oan to test this regex
+        const updatedText = originalText.replace(/^\s*\d+\./, (match) =>
+          match.replace(/\d+/, "1")
+        );
+        const textLength = token.line.length;
+        const lineNumber = token.lineNumber;
+
+        onError({
+          lineNumber: lineNumber,
+          detail: `\n  Expected: "${updatedText}"\n  Actual: "${originalText}"\n`,
+          fixInfo: {
+            lineNumber: lineNumber,
+            deleteCount: textLength,
+            insertText: updatedText,
```

Comment:
Ideally, we only delete the incorrect number and replace with `1` instead. By deleting the whole line and replacing it with the entire adjusted line, it clashes with other fixers.

For example:
- TOP003's fixers replace ordered lists in the LO section with an unordered list. But if TOP010 fixes via replacing the whole line, it causes only the first LO ordered list item to be converted, and the others remain untouched.
- If an ordered list item contains other errors within its text e.g. `_` for emphasis instead of `*`, then markdownlint can prioritise one of the rules then prevent the other from firing. From my testing, it prioritises fixing `_`, so that bit gets fixed but the list item prefix remains untouched.

e.g.
```md
1. Foo
2. _Bar_
```
gets "fixed" to
```md
1. Foo
2. *Bar*
```

The fix for this would be to count the number of leading spaces, get the actual incorrect prefix number (and its length so you know how many chars to delete), and use that info to only delete the number and replace with `1`. Just tested a very rudimentary implementation of this and can confirm this approach does not clash with other rules unlike wiping and replacing the entire line.